javascript-data-model/main.js

```diff
@@ -0,0 +1,28 @@
+/* exported orderHistory */
+
+var orderHistory = [
+  {
+    itemNames: ['JavaScript for Impatient Programmers'],
+    type: ['book'],
+    returnDeadline: 'September 7, 2020',
+    orderNumber: '114-394'
+  },
+  {
+    itemNames: ['The Timeless Way of Building'],
+    type: ['book'],
+    returnDeadline: 'August 19, 2020',
+    orderNumber: '113-998'
+  },
+  {
+    itemNames: ['Gamecube Controller Adapter'],
+    type: ['game acessories'],
+    returnDeadline: 'August 5, 2020',
+    orderNumber: '114-287'
+  },
+  {
+    itemNames: ['Gamecube Controller for Smash', 'The art of SQL'],
+    type: ['game acessories', 'book'],
+    returnDeadline: ['August 4, 2020', 'August 4, 2020'],
```

Comment:
But here, it is an array of strings. When the data type of the value differs like this between entries, looping over your data becomes a nightmare. Try to make sure that your data model and the types each value holds are the same between all entries. The data itself will vary, but the structure they are represented in should always be the same.